### PR TITLE
Fix issues caused by parallel deletion of parent-child resources

### PR DESCRIPTION
**Pull request by matthchr**
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

**Review by theunrepentantgeek — APPROVED** [review]
Approval GIF

File: v2/internal/testcommon/kube_per_test_context.go [hunk 1]
```diff
@@ -456,6 +456,42 @@ func (tc *KubePerTestContext) DeleteResourcesAndWait(objs ...client.Object) {
...
+
+	// Note: This is imperfect. If you have a situation like A owns B owns C and you pass just A and C here, we won't detect it
+	var roots []client.Object
+
+	for _, obj := range objs {
+		refs := obj.GetOwnerReferences()
+		foundOwner := false
+		for _, ref := range refs {
+			if _, ok := oids[ref.UID]; ok {
+				foundOwner = true
```
Comment (super-harsh) [inline v2/internal/testcommon/kube_per_test_context.go:484]: maybe `break` after when owner is found? As we don't want to keep iterating?

**Review by super-harsh — APPROVED** [review]

**Comment by matthchr** [comment]
This actually doesn't fix the problem fully, I need to iterate more on it.

File: v2/internal/testcommon/kube_per_test_context.go (hunk 1 above)
Comment (matthchr) [reply v2/internal/testcommon/kube_per_test_context.go:484]: Done

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/testcommon/kube_per_test_context.go [hunk 2]
```diff
@@ -461,18 +461,53 @@ func (tc *KubePerTestContext) DeleteResource(obj client.Object) {
...
 // DeleteResourcesAndWait deletes the resources in K8s and waits for them to be deleted
-func (tc *KubePerTestContext) DeleteResourcesAndWait(objs ...client.Object) {
+func (tc *KubePerTestContext) deleteResourcesAndWait(objs ...client.Object) {
 	for _, obj := range objs {
 		err := tc.kubeClient.Delete(tc.Ctx, obj)
 		err = client.IgnoreNotFound(err) // If the resource doesn't exist, that's good for us!
 		tc.G.Expect(err).To(gomega.Succeed())
 	}
 
+	// TODO: Okay so we have a problem here...
```
Comment (matthchr) [inline v2/internal/testcommon/kube_per_test_context.go:471]: Ah remove this, was meant to be notes for myself

**Pull request merged**